### PR TITLE
build: fix kafka connector integration test

### PR DESCRIPTION
**Pull request by chillleader**
## Description

This PR will fix the Kafka Integration test that has been failing since #2792 was merged.
- The test itself was adjusted to provide a custom retry policy, also a couple small errors were fixed in other test cases
- The consumer code now rethrows the exception when the retry loop fails (this shouldn't affect normal execution in production, as the retry loop is infinite there)

The test itself is marked as disabled, to be discussed whether we want to run it always.

## Related issues

closes https://github.com/camunda/team-connectors/issues/858

### CODE REVIEW

File: connectors/kafka/src/test/java/io/camunda/connector/kafka/integration/KafkaIntegrationTest.java [hunk 1]
```diff
@@ -448,7 +454,7 @@ void consumeMessageWithHeaders() {
             kafkaTopic,
             null,
             null,
-            List.of(0L, 0L),
+            null,
```
Comment (markfarkas-camunda) [inline connectors/kafka/src/test/java/io/camunda/connector/kafka/integration/KafkaIntegrationTest.java:457]: Just to clarify: Why was this change necessary?

**Review by markfarkas-camunda — COMMENTED** [review]
Looks good, I only have 1 question

**Review by sbuettner — APPROVED** [review]

**Review by johnBgood — APPROVED** [review]
thank you!

File: connectors/kafka/src/test/java/io/camunda/connector/kafka/integration/KafkaIntegrationTest.java (hunk 1 above)
Comment (chillleader) [reply connectors/kafka/src/test/java/io/camunda/connector/kafka/integration/KafkaIntegrationTest.java:457]: If we start reading from 0 again, wrong message will be consumed as we are running multiple tests in a certain order. I believe this was added by mistake.
